### ISSUE TITLE
API Proposal: Broadcasts with Layout

### ISSUE THREAD

**Issue by aoberoi** [opentok/Opentok-Java-SDK#116]
This is a request for comments regarding the API for upcoming platform functionality. The platform functionality is not generally available at the moment and there is no guarantee it will be released in any form.

this is related to #112 
### Create a Broadcast

``` java
// simplest case (`sessionId` is a String)
Future<Broadcast> broadcast = opentok.startBroadcast(sessionId);

// using predefined layout (`Layout.Pip` is an instance of `Layout`)
Future<Broadcast> broadcast = opentok.startBroadcast(sessionId, Layout.Pip);

// using a custom layout (`Layout.createCustom` returns an instance of `Layout`, `stylesheet` is a CSS string)
Future<Broadcast> broadcast = opentok.startBroadcast(sessionId, Layout.createCustom(stylesheet));
```
### Stop a Broadcast

``` java
// when you don't have a `Broadcast` instance (`broadcastId` is a String)
Future<Broadcast> broadcast = opentok.stopBroadcast(broadcastId);

Future<Broadcast> broadcast = broadcast.stop();
```
### Get a Broadcast

``` java
// `broadcastId` is a String
Future<Broadcast> broadcast = opentok.getBroadcast(broadcastId);

// `Broadcast` has the following read-only properties (getters)
String broadcastId = broadcast.getId();
String sessionId = broadcast.getSessionId();
Long broadcastCreatedAt = broadcast.getCreatedAt();
Long broadcastUpdatedAt = broadcast.getUpdatedAt();
String broadcastPartnerId = broadcast.getPartnerId();
Map<String, URL> broadcastUrls = broadcast.getUrls(); // the Map contains a key `"HLS"` with a value of a URL
```
### Get a Broadcast Layout

``` java
// when you don't have a `Broadcast` instance (`broadcastId` is a String)
Future<Layout> layout = opentok.getBroadcastLayout(broadcastId);

Future<Layout> layout = broadcast.getLayout()
```
### Update a Broadcast Layout

``` java
// when you don't have a `Broadcast` instance (`broadcastId` is a String, `layout` is an instance of `Layout`)
Future<Void> done = opentok.updateBroadcastLayout(broadcastId, layout);

// `layout` is an instance of `Layout`
Future<Void> done = broadcast.updateLayout(layout);
```
### Update a Stream's Layout Class List

Question: should we have an API for updating multiple streams?
Question: should these return a representation of the Stream properties?

``` java
String[] layoutClasses = { "class1" };

// when you don't have a `Session` instance (returns void, `sessionId` is a String)
opentok.updateStream(sessionId, streamId, new StreamOptions.Builder()
    .layoutClasses(layoutClasses)
    .build());

// returns void
session.updateStream(streamId, new StreamOptions.Builder()
    .layoutClasses(layoutClasses)
    .build());
```

**Comment by aoberoi** [comment]
we plan to begin an implementation next week. please comment at your earliest convenience.

**Comment by aoberoi** [comment]
It's important to answer this question now:

> Question: should these return a representation of the Stream properties?

if we want to have a different return type, that would be a breaking change.

**Comment by msach22** [comment]
This has been released with `4.3.0`
